src/script/components/VerificationBadge/VerificationBadges.tsx

```diff
@@ -98,7 +132,8 @@ export const DeviceVerificationBadges = ({
   device: ClientEntity;
   getIdentity?: (deviceId: string) => WireIdentity | undefined;
 }) => {
-  const MLSStatus = getIdentity?.(device.id)?.status;
+  const identity = getIdentity?.(device.id);
+  const {MLSStatus} = useMLSStatus(identity);
```

Comment:
maybe you can do the user lookup at this level (and avoid it in the `UserVerificationBadge`, since the user is already given there)